code/_helpers/filters.dm

```diff
@@ -0,0 +1,319 @@
+#define ICON_NOT_SET "Not Set"
+
+//This is stored as a nested list instead of datums or whatever because it json encodes nicely for usage in tgui
+GLOBAL_LIST_INIT(master_filter_info, list(
+	"alpha" = list(
+		"defaults" = list(
+			"x" = 0,
+			"y" = 0,
+			"icon" = ICON_NOT_SET,
+			"render_source" = "",
+			"flags" = 0
+		),
+		"flags" = list(
+			"MASK_INVERSE" = MASK_INVERSE,
+			"MASK_SWAP" = MASK_SWAP
+		)
+	),
+	"angular_blur" = list(
+		"defaults" = list(
+			"x" = 0,
+			"y" = 0,
+			"size" = 1
+		)
+	),
+	/* Not supported because making a proper matrix editor on the frontend would be a huge dick pain.
+		Uncomment if you ever implement it
+	"color" = list(
+		"defaults" = list(
+			"color" = matrix(),
+			"space" = FILTER_COLOR_RGB
+		)
+	),
+	*/
+	"displace" = list(
+		"defaults" = list(
+			"x" = 0,
+			"y" = 0,
+			"size" = null,
+			"icon" = ICON_NOT_SET,
+			"render_source" = ""
+		)
+	),
+	"drop_shadow" = list(
+		"defaults" = list(
+			"x" = 1,
+			"y" = -1,
+			"size" = 1,
+			"offset" = 0,
+			"color" = COLOR_HALF_TRANSPARENT_BLACK
+		)
+	),
+	"blur" = list(
+		"defaults" = list(
+			"size" = 1
+		)
+	),
+	"layer" = list(
+		"defaults" = list(
+			"x" = 0,
+			"y" = 0,
+			"icon" = ICON_NOT_SET,
+			"render_source" = "",
+			"flags" = FILTER_OVERLAY,
+			"color" = "",
+			"transform" = null,
+			"blend_mode" = BLEND_DEFAULT
+		)
+	),
+	"motion_blur" = list(
+		"defaults" = list(
+			"x" = 0,
+			"y" = 0
+		)
+	),
+	"outline" = list(
+		"defaults" = list(
+			"size" = 0,
+			"color" = COLOR_BLACK,
+			"flags" = NONE
+		),
+		"flags" = list(
```

Comment:
Does this really need to be a list? This looks like it can be made into a bitflag

Reply:
> Does this really need to be a list? This looks like it can be made into a bitflag

I agree with you, but at the moment for Byond it will be perceived as list strings. You can check this yourself by decoding the filter "outline".